rust/src/accountant.rs

```diff
@@ -0,0 +1,179 @@
+use crate::accumulator::{UsageAccumulator, UsageUnit};
+use crate::producer::{ClientError, KafkaConfig, KafkaProducer, Producer};
+use chrono::Local;
+use serde_json::json;
+
+static DEFAULT_TOPIC_NAME: &str = "shared-resources-usage";
+/// This is the entry point for the library. It is in most cases
+/// everything you need to instrument your application.
+///
+/// The UsageAccountant needs a `Producer` and a `UsageAccumulator`.
+/// Data is stored in the accumulator and, periodically, it is
+/// flushed into the Kafka topic via a consumer.
+///
+/// Accumulating data locally is critical to reduce the performance impact
+/// of this library to a minimum and reduce the amount of Kafka messages.
+/// This means that this structure should be instantiated rarely.
+/// Possibly only once per application (or per thread).
+///
+/// Avoid creating a UsageAccountant every time some data needs to
+/// be recorded.
+
+pub struct UsageAccountant<'a> {
+    accumulator: UsageAccumulator,
+    producer: Box<dyn Producer + 'a>,
```

Comment:
I tried that way, but then I had a problem in the unit test as I need to access the `messages` attribute of the `DummyProducer` mock object during tests.

If I make this static, I still have to pass a reference to a RefCell in teh producer initializer when I create the DummyProducer, otherwise I cannot do something like this https://github.com/getsentry/usage-accountant/pull/22/files#diff-b5869fbf23fdb0be35b0bdcc38b0985c55c8c83547ddcdae89c92cd147c9ca06R127.
But having to store the reference in the `DummyProducer` means The DummyProducer must have a lifetime, and if that is contained in the Accountant (which has no lifetime) the compiler complains in the unit test that `messages` does not live long enough. Here https://github.com/getsentry/usage-accountant/pull/22/files#diff-b5869fbf23fdb0be35b0bdcc38b0985c55c8c83547ddcdae89c92cd147c9ca06R121

I feel I might be doing something wrong. Do you know if there is a better way to inspect the content of a mock object (the producer) after I used it in the test. 
The DummyProducer is owned by the `Accountant` as a `Box<dyn Producer + 'a>,`. So I cannot get back a `DummyProducer` from the `Accountant`. Is there a way to cast `dyn Producer` into `DummyProducer` in the test. ?